datastream-server/src/main/java/com/linkedin/datastream/server/zk/ZkAdapter.java

```diff
@@ -827,6 +840,14 @@ public void addConnectorType(String connectorType) {
     return new HashSet<>(_zkclient.getChildren(KeyBuilder.connector(_cluster, connector)));
   }
 
+  @VisibleForTesting
+  Map<String, Set<String>> getAllConnectorTaskLocks(String connector) {
```

Comment:
This method is going to make N roundtrips to ZK where N's the number of unique task prefixes in the cluster. This number is close to 1500 in one of our bigger change capture clusters atm. Just calling this out in case extended lock cleanup duration could cause issues.